warehouse/manage/views/__init__.py

```diff
@@ -139,6 +140,79 @@
 from warehouse.utils.project import confirm_project, destroy_docs, remove_project
 
 
+class ManageAccountMixin:
+
+    def __init__(self, request):
+        self.request = request
+        self.user_service = request.find_service(IUserService, context=None)
+        self.breach_service = request.find_service(
+            IPasswordBreachedService, context=None
+        )
+
+    @view_config(request_method="POST", request_param=["reverify_email_id"])
+    def reverify_email(self):
+        try:
+            email = (
+                self.request.db.query(Email)
+                .filter(
+                    Email.id == int(self.request.POST["reverify_email_id"]),
+                    Email.user_id == self.request.user.id,
+                )
+                .one()
+            )
+        except NoResultFound:
+            self.request.session.flash("Email address not found", queue="error")
+            return self.default_response
+
+        if email.verified:
+            self.request.session.flash("Email is already verified", queue="error")
+        else:
+            verify_email_ratelimit = self.request.find_service(
+                IRateLimiter, name="email.verify"
+            )
+            if verify_email_ratelimit.test(self.request.user.id):
+                send_email_verification_email(self.request, (self.request.user, email))
+                verify_email_ratelimit.hit(self.request.user.id)
+                email.user.record_event(
+                    tag=EventTag.Account.EmailReverify,
+                    request=self.request,
+                    additional={"email": email.email},
+                )
+
+                self.request.session.flash(
+                    f"Verification email for {email.email} resent", queue="success"
+                )
+            else:
+                self.request.session.flash(
+                    (
+                        "Too many incomplete attempts to verify email address(es) for "
+                        f"{self.request.user.username}. Complete a pending "
+                        "verification or wait before attempting again."
+                    ),
+                    queue="error",
+                )
+
+        return HTTPSeeOther(self.request.route_path("manage.account"))
```

Comment:
Can we branch here? Redirect to `manage.account` if a user has a verified email and `manage.unverified-account` otherwise? The issue is it flashes the `You must verify your **primary** email address before you can perform this action.` when re-verifying from the unmanaged view.